subsys/bluetooth/host/hci_core.c

```diff
@@ -193,7 +193,8 @@ void bt_hci_host_num_completed_packets(struct net_buf *buf)
 		return;
 	}
 
-	if (!bt_conn_is_handle_valid(conn)) {
+	if (conn->state != BT_CONN_CONNECTED &&
+	    conn->state != BT_CONN_DISCONNECT) {
```

Comment:
This change here is effectively removing `BT_CONN_DISCONNECT_COMPLETE` from entering the clause, right? 

Can you elaborate on why the BT_CONN_DISCONNECT_COMPLETE should not be able to enter the clause?

Reply:
@LingaoM can you please reply?

Reply:
This is fixing a) in https://github.com/zephyrproject-rtos/zephyr/issues/38054#issuecomment-911518510